plaid/src/executor/mod.rs

```diff
@@ -19,13 +20,29 @@ use std::collections::HashMap;
 use std::sync::{Arc, RwLock};
 use std::thread::{self, JoinHandle};
 
-#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Serialize, Deserialize)]
 pub struct Message {
+    /// The message channel that this message is going to run on
     pub type_: String,
+    /// The data passed to the module
     pub data: Vec<u8>,
+    /// Any additional data that is provided as context to the module
+    /// This is usually either headers, or secrets
     pub accessory_data: HashMap<String, Vec<u8>>,
+    /// Where the message came from
     pub source: LogSource,
+    /// If this message is allowed to trigger additional messages to the same
+    /// or other message channels
     pub logbacks_allowed: LogbacksAllowed,
+    /// If a response is should be sent back to the source of the message
+    /// This is used in the GET request system to handle responses
+    #[serde(skip)]
+    pub response_sender: Option<OneShotSender<Option<String>>>,
```

Comment:
Could be nice to have this send a status code as well so that we're not always responding with a 200

Reply:
That would require the rule have an API with which to send a response code which it doesn't currently. Should we stub it out to add later or do you feel we should add such an API in this PR?

I'm somewhat against adding it in this PR because it will require rethinking how persistent response works in terms of caching which I'm not sure we need to tackle right now.

Reply:
Yeah that's a fair point - let's stub it out as something to add later. We have no use case for needing a status code right now so no reason to add complexity